Sources_v3/StreamChat/Database/DatabaseSession.swift

```diff
@@ -46,14 +46,16 @@ extension CurrentUserDatabaseSession {
 protocol MessageDatabaseSession {
     /// Creates a new `MessageDTO` object in the database. Throws an error if the message fails to be created.
     @discardableResult
-    func createNewMessage<ExtraData: MessageExtraData>(
+    func createNewMessage<ExtraData: ExtraDataTypes>(
         in cid: ChannelId,
         text: String,
         command: String?,
         arguments: String?,
         parentMessageId: MessageId?,
+        attachments: [_ChatMessageAttachment<ExtraData.Attachment>],
         showReplyInChannel: Bool,
-        extraData: ExtraData
+        extraData: ExtraData.Message,
+        extraDataType: ExtraData.Type
```

Comment:
Why do we need to specify `extraDataType` here when the whole function is generic over ExtraData?

Reply:
Moved discussion to Slack

Reply:
Conclusion: `_ChatMessageAttachment` is generic over `ExtraDataTypes` instead of `AttachmentExtraData` now.